snmp/datadog_checks/snmp/snmp.py

```diff
@@ -180,19 +176,15 @@ def fetch_results(self, config, all_oids, bulk_oids):
         for oid in bulk_oids:
             try:
                 self.log.debug('Running SNMP command getBulk on OID %r', oid)
-                binds_iterator = config.call_cmd(
-                    hlapi.bulkCmd,
+                binds = snmp_bulk(
+                    config,
+                    oid,
                     self._NON_REPEATERS,
                     self._MAX_REPETITIONS,
-                    oid,
-                    lookupMib=enforce_constraints,
-                    ignoreNonIncreasingOid=self.ignore_nonincreasing_oid,
-                    lexicographicMode=False,
```

Comment:
Is the `lexicographicMode` (which AFAIU means "please stay in the same sub-tree as the requested OIDs/don't make hoops across the OID tree") handled by this new implementation?

Reply:
Yes, this is implemented with the `isPrefixOf` calls. There is no flag anymore because we always want that behavior in our case.